include/drivers/gpio.h

```diff
@@ -55,12 +55,19 @@ extern "C" {
 /* Initializes output to a high state. */
 #define GPIO_OUTPUT_INIT_HIGH   (1U << 11)
 
+/* Initializes output based on logic level */
+#define GPIO_OUTPUT_INIT_LOGICAL (1U << 23)
```

Comment:
We could use bit 9 for the flag and move `GPIO_INPUT` and `GPIO_OUTPUT` down by one.

Reply:
I like that the current arrangement leaves the low 8 bits for flags that come from device-tree, and I'd like to preserve that in anticipation of encoding a `gpios` property as a 4-byte values (two-byte controller handle, pin number, dts flags).  So I just shifted everything up one bit.